src/lexer/jsx.ts

```diff
@@ -64,12 +63,48 @@ export function scanJSXToken(parser: ParserState, context: Context): Token {
   switch (token) {
     // '<'
     case Token.LessThan: {
-      advanceChar(parser);
-      if (parser.currentChar === Chars.Slash) {
-        advanceChar(parser);
-        parser.setToken(Token.JSXClose);
+      nextToken(parser, context);
+      if (parser.getToken() === Token.JSXClose) break;
+      if (parser.getToken() !== Token.LessThan)
+        report(parser, Errors.UnexpectedToken, KeywordDescTable[parser.getToken() & Token.Type]);
+
+      // Possible whitespace or comments between "<" and "/".
+      const {
+        index,
+        line,
+        column,
+        tokenIndex,
+        tokenColumn,
+        tokenLine,
+        startIndex,
+        startColumn,
+        startLine,
+        currentChar,
+        tokenValue,
+        tokenRaw,
+        tokenRegExp
+      } = parser;
+      const tokenAfter = scanSingleToken(parser, context, LexerState.None);
+      parser.tokenIndex = tokenIndex;
+      parser.tokenColumn = tokenColumn;
+      parser.tokenLine = tokenLine;
+      parser.startIndex = startIndex;
+      parser.startColumn = startColumn;
+      parser.startLine = startLine;
+      // Punctuator doesn't change tokenValue or raw, just keep previous
+      parser.tokenValue = tokenValue;
+      parser.tokenRaw = tokenRaw;
+      parser.tokenRegExp = tokenRegExp;
+
+      if (tokenAfter === Token.Divide) {
+        // Rewrite LessThan + Divide as JSXClose
+        parser.setToken(Token.JSXClose, true);
       } else {
-        parser.setToken(Token.LessThan);
+        // Restore if not merged
+        parser.index = index;
+        parser.line = line;
+        parser.column = column;
+        parser.currentChar = currentChar;
```

Comment:
It's not a simple restore if not merged. Even for merged, has to take care of token start position, otherwise onToken will miss all chars before "/".

The other choice is to keep "< /" as two tokens "<" and "/", that's natural, but require some refactoring of jsx parsing. Might be better.